blog/_posts/2025-04-10-infiniband-on-aks.md

```diff
@@ -0,0 +1,45 @@
+---
+title: "Simplifying InfiniBand on AKS"
+description: "Learn the what, whys and hows of configuring InfiniBand networking for high performance compute (HPC) workloads on AKS"
+date: 2025-04-10
+authors: Sachi Desai, Suraj Deshmukh, Ernest Wong
+categories: 
+- AI
+- networking
+---
+
+## What is InfiniBand?
+
+High performance computing (HPC) workloads, like large-scale distributed AI training and inferencing, often require fast, reliable data transfer and synchronization across the underlying compute. Model training, for example, requires shared memory across GPUs because the parameters and gradients need to be constantly shared. For models with billions of parameters, the available memory in a single GPU node may not be enough, so "pooling" the memory across multiple nodes also requires high memory bandwidth due to the sheer volume of data involved. A common way to achieve this at scale is with a high-speed, low-latency network interconnect technology called **InfiniBand (IB)**. 
+
+Imagine InfiniBand networking as an incredibly fast highway for data transfer, and the data as cars that need to travel from one city to another. On a typical highway (like traditional IP networks), cars must follow speed limits, obey traffic signals, and sometimes stop in traffic jams, which slows them down.
+InfiniBand, on the other hand, can be considered a special highway built just for race cars - it has no speed limits, no traffic lights, and wide lanes, allowing the cars to zoom at top speed without any interruptions. This makes data travel incredibly fast and efficiently.
+
+There are two ways to use this fast highway:
+- **Remote Direct Memory Access (RDMA)**: Similar to driving a race car on the race car highway. It maximizes speed and performance but may require specific application design and networking configuration to drive these race cars.
+- **IP over InfiniBand (IPoIB)**: This is comparable to regular cars using the race car highway - may be easy to implement and compatible with off-the-shelf applications, but you don't get the full speed benefits.
+
+Choosing between these two approaches depends on whether you need compatibility and ease (like regular cars on a race car highway) or top-notch speed and performance (like race cars on a race car highway).
+
+## RDMA over InfiniBand versus IPoIB

```

Comment:
Is this different from just RDMA above? Should be consistent with the naming if you're talking about the same thing: "RDMA over Infiniband" or just "RDMA"